packages/next/build/webpack-config.ts

```diff
@@ -176,6 +177,109 @@ export default async function getBaseWebpackConfig(
 
   const devtool = dev ? 'cheap-module-source-map' : false
 
+  // Contains various versions of the Webpack SplitChunksPlugin used in different build types
+  const splitChunksConfigs: {
+    [propName: string]: webpack.Options.SplitChunksOptions
+  } = {
+    dev: {
+      cacheGroups: {
+        default: false,
+        vendors: false,
+      },
+    },
+    selective: {
+      cacheGroups: {
+        default: false,
+        vendors: false,
+        react: {
+          name: 'commons',
+          chunks: 'all',
+          test: /[\\/]node_modules[\\/](react|react-dom)[\\/]/,
+        },
+      },
+    },
+    prod: {
+      chunks: 'all',
+      cacheGroups: {
+        default: false,
+        vendors: false,
+        commons: {
+          name: 'commons',
+          chunks: 'all',
+          minChunks: totalPages > 2 ? totalPages * 0.5 : 2,
+        },
+        react: {
+          name: 'commons',
+          chunks: 'all',
+          test: /[\\/]node_modules[\\/](react|react-dom)[\\/]/,
+        },
+      },
+    },
+    prodGranular: {
+      chunks: 'all',
+      cacheGroups: {
+        default: false,
+        vendors: false,
+        framework: {
+          name: 'framework',
+          chunks: 'all',
+          test: /[\\/]node_modules[\\/](react|react-dom|scheduler|prop-types)[\\/]/,
+          priority: 40,
+        },
+        lib: {
+          test(module: { size: Function }): boolean {
+            return module.size() > 160000
+          },
+          name(module: { identifier: Function }): string {
+            let nodeModuleNameResults = /(?:^|\/)node_modules\/(.*)/.exec(
+              module.identifier()
+            )
+            return nodeModuleNameResults
+              ? nodeModuleNameResults[1]
```

Comment:
To avoid nested directories when a rawRequest is a scoped module, you might want to "dasherize" the result.
```suggestion
              ? nodeModuleNameResults[1].replace(/^@(\w+)\//, '$1-')
```
For example, when the `rawRequest` is `'@scope/module-name'`:
```diff
-.next/static/chunks/@scope/module-name.{contentHash}.js
+.next/static/chunks/scope-module-name.{contentHash}.js
```